src/mlpack/methods/ann/layer/celu.hpp

```diff
@@ -0,0 +1,203 @@
+ *   \begin{array}{lr}
+ *     1 & : x >= 0 \\
+ *     (f(x)/alpha) + 1 & : x \le 0
+ *   \end{array}
+ * \right.
+ * @f}
+ *
+ * For more information, read the following paper:
+ *
+ * @code
+ * @article{
+ *   author  = {Jonathan T. Barron},
+ *   title   = {Continuously Differentiable Exponential Linear Units},
+ *   year    = {2017}
+ * }
+ * @endcode
+ *
+ * In the deterministic mode, there is no computation of the derivative.
+ *
+ * @note During training deterministic should be set to false and during
+ *       testing/inference deterministic should be set to true.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class CELU
+{
+ public:
+  /**
+   * Create the CELU object using the specified parameter. The non zero
+   * gradient for negative inputs can be adjusted by specifying the CELU
+   * hyperparameter alpha (alpha > 0).
+   *
+   * @param alpha Scale parameter for the negative factor.
+   */
+  CELU(const double alpha);
+
+  /**
+   * Ordinary feed forward pass of a neural network, evaluating the function
+   * f(x) by propagating the activity forward through f.
+   *
+   * @param input Input data used for evaluating the specified function.
+   * @param output Resulting output activation.
+   */
+	template<typename InputType, typename OutputType>
+	void Forward(const InputType&& input, OutputType&& output);
+
+  /**
+   * Ordinary feed backward pass of a neural network, calculating the function
+   * f(x) by propagating x backwards through f. Using the results from the feed
+   * forward pass.
+   *
+   * @param input The propagated input activation f(x).
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient.
+   */
+	template<typename DataType>
+	void Backward(const DataType&& input, DataType&& gy, DataType&& g);
+	
+  //! Get the output parameter.
+	OutputDataType const& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+	OutputDataType& OutputParameter() { return outputParameter; }
+	
+  //! Get the delta.
+	OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+	OutputDataType& Delta() { return delta; }
+
+  //! Get the non zero gradient.
+	double const& Alpha() const { return alpha; }
+  //! Modify the non zero gradient.
+	double& Alpha() { return alpha; }
+
+  /**
+   * Serialize the layer.
+   */
+	template<typename Archive>
+	void serialize(Archive& ar, const unsigned int /* version */);
+
+ private:
+  /**
+   * Computes the value of activation function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  double Fn(const double x)
+  {
+    if(x < DBL_MAX)
+    {
+      return (x >= 0) ? x : alpha * (std::exp((x / alpha) - 1));
```

Comment:
I believe the -1 is not part of the exponentiation (from what I see from the paper).
Therefore I think the right expression would be:
`return (x >= 0) ? x : alpha * (std::exp(x / alpha) - 1);`